python/server.py

```diff
@@ -66,256 +107,371 @@ def empty_to_none(field):
+                webhook='https://www.example.com',
+                client_report_id='123',
+                user=AssetReportUser(
+                    client_user_id='789',
+                    first_name='Jane',
+                    middle_name='Leah',
+                    last_name='Doe',
+                    ssn='123-45-6789',
+                    phone_number='(555) 123-4567',
+                    email='jane.doe@example.com',
+                )
+            )
+        )
+
+        asset_report_create_response = client.asset_report_create(request)
+        print(asset_report_create_response)
+        asset_report_token = asset_report_create_response['asset_report_token']
+    except plaid.ApiException as e:
+        error_response = format_error(e)
+        return jsonify(error_response)
+
+    # Poll for the completion of the Asset Report.
+    num_retries_remaining = 20
+    asset_report_json = None
+    while num_retries_remaining > 0:
+        try:
+            request = AssetReportGetRequest(
+                asset_report_token=asset_report_token,
+            )
+            asset_report_get_response = client.asset_report_get(request)
+            asset_report_json = asset_report_get_response['report']
+            break
+        except plaid.ApiException as e:
+            response = json.loads(e.body)
+            if response['error_code'] == 'PRODUCT_NOT_READY':
+                num_retries_remaining -= 1
+                time.sleep(1)
+                continue
+        error_response = format_error(e)
+        return jsonify(error_response)
+    if asset_report_json is None:
+        return jsonify({'error': {'status_code': e.status, 'display_message':
+                                  'Timed out when polling for Asset Report', 'error_code': '', 'error_type': ''}})
+
+    asset_report_pdf = None
+    try:
+        pdf_request = AssetReportPDFGetRequest(
+            asset_report_token=asset_report_token,
+        )
+        pdf = client.asset_report_pdf_get(pdf_request)
+        return jsonify({
+            'error': None,
+            'json': asset_report_json.to_dict(),
+            'pdf': base64.b64encode(pdf.read()).decode('utf-8'),
+        })
+    except plaid.ApiException as e:
+        error_response = format_error(e)
+        return jsonify(error_response)
+
 
 # Retrieve investment holdings data for an Item
 # https://plaid.com/docs/#investments
+
+
 @app.route('/api/holdings', methods=['GET'])
 def get_holdings():
-  try:
-    holdings_response = client.Holdings.get(access_token)
-  except plaid.errors.PlaidError as e:
-    return jsonify({'error': {'display_message': e.display_message, 'error_code': e.code, 'error_type': e.type } })
-  pretty_print_response(holdings_response)
-  return jsonify({'error': None, 'holdings': holdings_response})
+    try:
+        h_request = InvestmentsHoldingsGetRequest(access_token=access_token)
+        holdings_response = client.investments_holdings_get(h_request)
```

Comment:
we have a strange pattern throughout this code where we have `h_request` and `holdings_response`, or another example `ag_request` and `auth_response`, and in other places simply `request` and `response`. Can we use the same naming scheme for all requests and responses? I dont care which we use, `request`, `ag_request`, or `auth_get_request` etc but lets just stick to one.

Reply:
cleaned up request and responses for all backends to make them consistent within their languages.